src/coprocessor/dag/expr/ctx.rs

```diff
@@ -47,6 +47,136 @@ pub const FLAG_DIVIDED_BY_ZERO_AS_WARNING: u64 = 1 << 8;
 pub const MODE_ERROR_FOR_DIVISION_BY_ZERO: u64 = 27;
 
 const DEFAULT_MAX_WARNING_CNT: usize = 64;
+
+pub struct EvalConfigBuilder {
+    /// timezone
+    tz: FixedOffset,
+    ignore_truncate: bool,
+    truncate_as_warning: bool,
+    overflow_as_warning: bool,
+    in_insert_stmt: bool,
+    in_update_or_delete_stmt: bool,
+    in_select_stmt: bool,
+    pad_char_to_full_length: bool,
+    divided_by_zero_as_warning: bool,
+    max_warning_cnt: usize,
+    sql_mode: u64,
+    strict_sql_mode: bool,
+}
+
+impl EvalConfigBuilder {
+    pub fn new() -> EvalConfigBuilder {
+        EvalConfigBuilder {
+            tz: FixedOffset::east(0),
+            ignore_truncate: false,
+            truncate_as_warning: false,
+            overflow_as_warning: false,
+            in_insert_stmt: false,
+            in_update_or_delete_stmt: false,
+            in_select_stmt: false,
+            pad_char_to_full_length: false,
+            divided_by_zero_as_warning: false,
+            max_warning_cnt: DEFAULT_MAX_WARNING_CNT,
+            sql_mode: 0,
+            strict_sql_mode: false,
+        }
+    }
+
+    pub fn set_timezone(&mut self, tz_offset: i64) -> Result<&mut Self> {
```

Comment:
What do you think about checking time zones in the build method? So it has consistent setter APIs

Reply:
+1. We can save the offset and name, and check it in `build()`.

Reply:
Agreed, I'll edit this later (very busy ATM)

Reply:
@overvenus A problem: if you don't set three timezone, it should be always correct since the default value is correct. But if we check the timezone in `build` we still have to call an `unwrap`, which is very annoying. What do you think about this?

Reply:
This is why I'm putting the timezone check in `set_timezone`

Reply:
I think it's ok, because this case is rare and we only use it in tests.

Reply:
I am okay with `fn set_timezone() -> Result<()>`.

Reply:
@breeswish Seems that the originator of this idea has changed his mind. What about you?